crates/ruff_linter/src/rules/ruff/rules/unnecessary_regular_expression.rs

```diff
@@ -0,0 +1,243 @@
+    fn compare_expr(&self, op: CmpOp) -> Expr {
+        Expr::Compare(ExprCompare {
+            left: Box::new(self.pattern.clone()),
+            ops: Box::new([op]),
+            comparators: Box::new([self.string.clone()]),
+            range: TextRange::default(),
+        })
+    }
+
+    /// Return a new method call expression on `self.string` with `args` like
+    /// `self.string.method(args...)`
+    fn method_expr(&self, method: &str, args: Vec<Expr>) -> Expr {
+        let method = Expr::Attribute(ExprAttribute {
+            value: Box::new(self.string.clone()),
+            attr: Identifier::new(method, TextRange::default()),
+            ctx: ExprContext::Load,
+            range: TextRange::default(),
+        });
+        Expr::Call(ExprCall {
+            func: Box::new(method),
+            arguments: Arguments {
+                args: args.into_boxed_slice(),
+                keywords: Box::new([]),
+                range: TextRange::default(),
+            },
+            range: TextRange::default(),
+        })
+    }
+}
+
+/// RUF055
+pub(crate) fn unnecessary_regular_expression(checker: &mut Checker, call: &ExprCall) {
+    // adapted from unraw_re_pattern
+    let semantic = checker.semantic();
+
+    if !semantic.seen_module(Modules::RE) {
+        return;
+    }
+
+    let Some(qualified_name) = semantic.resolve_qualified_name(call.func.as_ref()) else {
+        return;
+    };
+
+    let ["re", func] = qualified_name.segments() else {
+        return;
+    };
+
+    // skip calls with more than `pattern` and `string` arguments (and `repl`
+    // for `sub`)
+    let Some(re_func) = ReFunc::from_call_expr(checker, call, func) else {
+        return;
+    };
+
+    let Some(pat) = call.arguments.find_argument("pattern", 0) else {
+        // this should be unreachable given the checks above, so it might be
+        // safe to unwrap here instead
+        return;
+    };
+
+    // For now, restrict this rule to string literals
+    let Some(string_lit) = pat.as_string_literal_expr() else {
+        return;
+    };
+
+    // For now, reject any regex metacharacters. Compare to the complete list
+    // from https://docs.python.org/3/howto/regex.html#matching-characters
+    let has_metacharacters = string_lit.value.chars().any(|c| {
+        matches!(
+            c,
+            '.' | '^' | '$' | '*' | '+' | '?' | '{' | '}' | '[' | ']' | '\\' | '|' | '(' | ')'
+        )
+    });
+
+    if has_metacharacters {
+        return;
+    }
+
+    // Here we know the pattern is a string literal with no metacharacters, so
+    // we can proceed with the str method replacement
+    let new_expr = re_func.replacement();
+
+    let repl = checker.generator().expr(&new_expr);
+    let mut diagnostic = Diagnostic::new(
+        UnnecessaryRegularExpression {
+            replacement: repl.clone(),
+        },
+        call.range,
+    );
+
+    diagnostic.set_fix(Fix::safe_edit(Edit::replacement(
```

Comment:
We should mark the fix as unsafe if the `call.range` contains any comments. You can use `checker.comment_ranges.has_comments` for that. 

We should then extend the documentation to explain the rules fixability